File: 백준 문제 풀이 모음/python/class 3/11723.py

import sys

m = int(sys.stdin.readline())

class Set() :
    def __init__ (self) :
        self.l =[]
    
    def add(self, x) :
        if x not in self.l :
            self.l.append(x)

    def remove(self, x) :
        if x in self.l :
            self.l.remove(x)

    def check(self, x) :
        if x in self.l :
            return 1
        else :  
            return 0

    def all (self) :
        self.l = ['1','2','3','4','5','6','7','8','9','10','11','12','13','14','15','16','17','18','19','20']
    
    def toggle (self, x) :
        if x in self.l :
            self.l.remove(x)
        else :
            self.l.append(x)
    
    def empty (self) :
        self.l= []

    def show (self) :
        return self.l

model = Set()

for _ in range(m) :
    x = sys.stdin.readline().rstrip().split()

    if x[0] == 'add' :
        model.add(x[1])
    
    elif x[0] == 'remove' : 
        model.remove(x[1])
    
    elif x[0] == 'check' :
        print(model.check(x[1]))
        
    
    elif x[0] == 'toggle' :
        model.toggle(x[1])
    
    elif x[0] == 'all' :
        model.all()

    elif x[0] == 'empty' :
        model.empty()

File: 백준 문제 풀이 모음/kotlin/class 3/11723.kt


fun main() = with(System.`in`.bufferedReader()){
    val n = readLine().toInt()
    val list = MutableList<Int>(21) { 0 }
    val sb = StringBuilder()

    repeat(n) {
        val input = readLine().split(' ').toList()

        when(input[0]) {
            "add" -> {
                val cost = input[1].toInt()
                list[cost] = 1
            }
            "remove" -> {
                val cost = input[1].toInt()
                if (list[cost] == 1) {
                    list[cost] = 0
                }
            }
            "check" -> {
                val cost = input[1].toInt()
                if (list[cost] == 1) sb.append(1)
                else sb.append(0)

                sb.append('\n')
            }
            "toggle" -> {
                val cost = input[1].toInt()
                if (list[cost] == 1) list[cost] = 0
                else list[cost] = 1
            }
            "all" -> {
                for(i in 1 .. 20) {
                    list[i] = 1
                }
            }
           "empty" -> {
               for(i in 1 .. 20) {
                   list[i] = 0
               }
           }
        }

    }

    println(sb.toString())

}
